### PR TITLE
Add a signal for when the CDC control state changes

### PR DESCRIPTION
**Pull request by m5p3nc3r**
This allows an async wait for changes in dts for example.

### CODE REVIEW

**Comment by Dirbaio** [comment]
Directly exposing a Signal allows the user to do other stuff with it, like signal it. We want the user to only be able to wait. 

The public API should be methods on `CdcAcmClass` like `wait_for_control_lines_change` (or concrete ones like  `wait_for_dtr_high` if you prefer).

File: embassy-usb/src/class/cdc_acm.rs [hunk 1]
```diff
@@ -30,6 +32,9 @@ const REQ_SET_LINE_CODING: u8 = 0x20;
 const REQ_GET_LINE_CODING: u8 = 0x21;
 const REQ_SET_CONTROL_LINE_STATE: u8 = 0x22;
 
+/// Signal will be send whenever the CDC control state changes
+pub static CDC_CONTROL_CHANGED: Signal<CriticalSectionRawMutex, ()> = Signal::new();
```
[the remark this answers is not in the document]
Comment (m5p3nc3r) [reply embassy-usb/src/class/cdc_acm.rs:36]: Ok, didn't realise it could be instantiated multiple times.  Will have a go at refactoring.

**Comment by m5p3nc3r** [comment]
I have updated this to use a WakerRegistration now.  Ready for review again.

Sorry, but my git-fu is horribly lacking and my attempt to squash the commits failed.  I assume you can do that on merge?

**Review by Dirbaio — APPROVED** [review]
lgtm, thank you! :)

**Pull request merged**